worlds/sc2/__init__.py

```diff
@@ -136,293 +174,386 @@ def setup_events(player: int, locked_locations: typing.List[str], location_cache
+            basic_units = {ItemNames.REAPER, ItemNames.GOLIATH, ItemNames.SIEGE_TANK, ItemNames.VIKING, ItemNames.BANSHEE}
+        basic_unit_options = [
+            item for item in possible_starter_items.values()
+            if item.name in basic_units
+            and ItemFilterFlags.StartInventory not in item.flags
+        ]
+        
+        # For Sudden Strike, starter units need an upgrade to help them get around
+        nco_support_items = {
+            ItemNames.REAPER: ItemNames.REAPER_SPIDER_MINES,
+            ItemNames.GOLIATH: ItemNames.GOLIATH_JUMP_JETS,
+            ItemNames.SIEGE_TANK: ItemNames.SIEGE_TANK_JUMP_JETS,
+            ItemNames.VIKING: ItemNames.VIKING_SMART_SERVOS,
+        }
+        if first_mission == SC2Mission.SUDDEN_STRIKE:
+            basic_unit_options = [
+                item for item in basic_unit_options
+                if item.name not in nco_support_items
+                or nco_support_items[item.name] in possible_starter_items
+                and ((ItemFilterFlags.Plando|ItemFilterFlags.Excluded) & possible_starter_items[nco_support_items[item.name]].flags) == 0
+            ]
+        if not basic_unit_options:
+            raise Exception("Early Unit: At least one basic unit must be included")
+        local_basic_unit = [item for item in basic_unit_options if ItemFilterFlags.NonLocal not in item.flags]
+        if local_basic_unit:
+            basic_unit_options = local_basic_unit
+        
+        unit = world.random.choice(basic_unit_options)
+        unit.flags |= ItemFilterFlags.StartInventory
+
+        # NCO-only specific rules
+        if first_mission == SC2Mission.SUDDEN_STRIKE:
+            if unit.name in nco_support_items:
+                support_item = possible_starter_items[nco_support_items[unit.name]]
+                support_item.flags |= ItemFilterFlags.StartInventory
+            if ItemNames.NOVA_JUMP_SUIT_MODULE in possible_starter_items:
+                possible_starter_items[ItemNames.NOVA_JUMP_SUIT_MODULE].flags |= ItemFilterFlags.StartInventory
+        if MissionFlag.Nova in first_mission.flags:
+            possible_starter_weapons = (
+                ItemNames.NOVA_HELLFIRE_SHOTGUN,
+                ItemNames.NOVA_PLASMA_RIFLE,
+                ItemNames.NOVA_PULSE_GRENADES,
+            )
+            starter_weapon_options = [item for item in possible_starter_items.values() if item.name in possible_starter_weapons]
+            starter_weapon = world.random.choice(starter_weapon_options)
+            starter_weapon.flags |= ItemFilterFlags.StartInventory
+        mission_count = sum(len(campaign) for campaign in world.mission_req_table.values())
+        if mission_count <= 10 and world.final_mission_id == SC2Mission.END_GAME.id:
+            if ItemNames.LIBERATOR_RAID_ARTILLERY in possible_starter_items:
+                possible_starter_items[ItemNames.LIBERATOR_RAID_ARTILLERY].flags |= ItemFilterFlags.StartInventory
```

Comment:
This is mainly there for possible starter Flashpoint, not early End Game. End Game can be excluded and the player still might start with Flashpoint

Reply:
Flashpoint is classed as a hard mission, I don't see it as very likely that it would be a starter mission unless the player has very deliberately made a yaml to make that likely. If they did that, I would expect them to also handle setting the logic level and start inventory such that they could beat the seed.

Do you have specific settings that would cause a generation failure without this adjustment? If so, I could make a unit test to guarantee we solve the problem. Otherwise, I'm inclined to just remove this adjustment.

Reply:
Made a usecase test for NCO only with no-builds excluded, and the test passed fine with this code snippet removed. I'm removing it fully.

Reply:
Flashpoint is one of possible starters:
```python
    if len(mission_pools[MissionPools.STARTER]) + len(mission_pools[MissionPools.EASY]) < 2:
        # Flashpoint needs just a few items at start but competent comp at the end
        move_mission(SC2Mission.FLASHPOINT, MissionPools.HARD, MissionPools.EASY)
```
The order is vanilla shuffled and the run is NCO only. (the NCO lane doesn't have starter a mission but starts with an easy one)